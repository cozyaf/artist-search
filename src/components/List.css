.List {
  ul {
    list-style: none;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin: 0;
  
  }
  li {
    border-radius: 5px;
    box-shadow: 0px 14px 60px rgba(20, 21, 34, 0.2);
    width: 250px;
    max-height: 200px;
    margin: 25px;
    padding: 30px;
  }
  p {
    font-size: .7em;
    overflow-wrap: break-word;
  }
}