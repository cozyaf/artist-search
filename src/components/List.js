import React from 'react';
import PropTypes from 'prop-types';

function List({ list, keyName, ListItem }) {
  const listElements = list.map(item => {
    const props = {
      [keyName]: item
    };
    return (
      <li key={item.id}>
        <ListItem {...props} />
      </li>
    );
  });

  return (
    <section>
      <ul>IM ALIVE{listElements}</ul>
    </section>
  );
}

List.propTypes = {
  list: PropTypes.array.isRequired,
  ListItem: PropTypes.func.isRequired,
  keyName: PropTypes.string.isRequired
};

export default List;


