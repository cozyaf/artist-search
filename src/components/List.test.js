import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List list={[{}]} keyName="" ListItem={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
