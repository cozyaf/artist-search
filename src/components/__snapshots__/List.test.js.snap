// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<section
  className="List"
>
  <ul>
    <li>
      <ListItem
        ={Object {}}
      />
    </li>
  </ul>
</section>
`;
